flink-connectors/flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/table/Elasticsearch6DynamicSink.java

```diff
@@ -128,9 +128,9 @@ public SinkFunctionProvider getSinkRuntimeProvider(Context context) {
 				upsertFunction);
 
 			builder.setFailureHandler(config.getFailureHandler());
-			config.getBulkFlushMaxActions().ifPresent(builder::setBulkFlushMaxActions);
-			config.getBulkFlushMaxSize().ifPresent(builder::setBulkFlushMaxSizeMb);
-			config.getBulkFlushInterval().ifPresent(builder::setBulkFlushInterval);
+			builder.setBulkFlushMaxActions(config.getBulkFlushMaxActions());
+			builder.setBulkFlushMaxSizeMb((int) (config.getBulkFlushMaxByteSize() >> 20));
```

Comment:
I wanted to have any formatting/transformation in one place in the `ElasticsearchOptions`. Can we return Mb from `config.getBulkFlushMaxByteSize`?

Reply:
I though about this, however, I think we should validate the value is in MB granularity. If we remove the remainder, it will be harder to check it.

Reply:
Then return the `MemorySize` instead? Without copying the shifting logic which is implemented inside of the `MemorySize`.

Reply:
But we have to move the `-1` logic out of it then.

Reply:
right, let's leave it as it is then.